Nix CI failed since ffi 1.16.0 released

https://github.com/kachick/ruby-ulid/actions/runs/6287353596/job/17071798079#step:8:188
https://rubygems.org/gems/ffi/versions/1.16.0
https://github.com/ffi/ffi/releases/tag/v1.16.0

No issues found https://github.com/ffi/ffi/issues

```
checking for whether -Wl,--exclude-libs,ALL is accepted as LDFLAGS... yes
checking for rb_gc_mark_movable()... yes
checking for whether -pthread is accepted as LDFLAGS... yes
creating extconf.h
creating Makefile

current directory:
/home/runner/work/ruby-ulid/ruby-ulid/vendor/bundle/ruby/3.2.0/gems/ffi-1.16.0/ext/ffi_c
make DESTDIR\= sitearchdir\=./.gem.20230924-16715-kcuzn1
sitelibdir\=./.gem.20230924-16715-kcuzn1 clean

current directory:
/home/runner/work/ruby-ulid/ruby-ulid/vendor/bundle/ruby/3.2.0/gems/ffi-1.16.0/ext/ffi_c
make DESTDIR\= sitearchdir\=./.gem.20230924-16715-kcuzn1
sitelibdir\=./.gem.20230924-16715-kcuzn1
Configuring libffi
configure: error: cannot find required auxiliary files: compile
make: *** [libffi.mk:8:
"/home/runner/work/ruby-ulid/ruby-ulid/vendor/bundle/ruby/3.2.0/gems/ffi-1.16.0/ext/ffi_c/libffi-x86_64-linux"/.libs/libffi_convenience.a]
Error 1

make failed, exit code 2

Gem files will remain installed in
/home/runner/work/ruby-ulid/ruby-ulid/vendor/bundle/ruby/3.2.0/gems/ffi-1.16.0
for inspection.
Results logged to
/home/runner/work/ruby-ulid/ruby-ulid/vendor/bundle/ruby/3.2.0/extensions/x86_64-linux/3.2.0/ffi-1.16.0/gem_make.out

/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/rubygems/ext/builder.rb:[119](https://github.com/kachick/ruby-ulid/actions/runs/6287353596/job/17071798079#step:8:120):in
`run'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/rubygems/ext/builder.rb:53:in
`block in make'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/rubygems/ext/builder.rb:45:in
`each'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/rubygems/ext/builder.rb:45:in
`make'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/rubygems/ext/ext_conf_builder.rb:42:in
`build'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/rubygems/ext/builder.rb:187:in
`build_extension'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/rubygems/ext/builder.rb:221:in
`block in build_extensions'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/rubygems/ext/builder.rb:218:in
`each'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/rubygems/ext/builder.rb:218:in
`build_extensions'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/rubygems/installer.rb:846:in
`build_extensions'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/bundler/rubygems_gem_installer.rb:72:in
`build_extensions'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/bundler/rubygems_gem_installer.rb:28:in
`install'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/bundler/source/rubygems.rb:200:in
`install'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/bundler/installer/gem_installer.rb:54:in
`install'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/bundler/installer/gem_installer.rb:16:in
`install_from_spec'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/bundler/installer/parallel_installer.rb:156:in
`do_install'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/bundler/installer/parallel_installer.rb:[147](https://github.com/kachick/ruby-ulid/actions/runs/6287353596/job/17071798079#step:8:148):in
`block in worker_pool'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/bundler/worker.rb:62:in
`apply_func'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/bundler/worker.rb:57:in
`block in process_queue'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/bundler/worker.rb:54:in
`loop'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/bundler/worker.rb:54:in
`process_queue'
/nix/store/naq4gk38gpykbfkwdjhs5a7wl58kl2yf-ruby-3.2.2/lib/ruby/3.2.0/bundler/worker.rb:90:in
`block (2 levels) in create_threads'

An error occurred while installing ffi (1.16.0), and Bundler cannot continue.

In Gemfile:
  steep was resolved to 1.5.3, which depends on
    listen was resolved to 3.8.0, which depends on
      rb-inotify was resolved to 0.10.1, which depends on
        ffi
Error: Process completed with exit code 5.
```

--- a/.github/workflows/ci-nix.yml
+++ b/.github/workflows/ci-nix.yml
@@ -26,7 +26,7 @@ jobs:
       matrix:
         os:
           - ubuntu-latest
-          - macos-latest
+          - macos-13
     runs-on: ${{ matrix.os }}
     timeout-minutes: 30
     steps:

--- a/.github/workflows/ci-nix.yml
+++ b/.github/workflows/ci-nix.yml
@@ -21,7 +21,13 @@ on:
 
 jobs:
   tasks:
-    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
+    runs-on: ${{ matrix.os }}
     timeout-minutes: 30
     steps:
       - uses: actions/checkout@v4

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -35,7 +35,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       # Keep same version as used in *.nix
-      - uses: crate-ci/typos@v1.16.8
+      - uses: crate-ci/typos@v1.16.12
         with:
           # https://github.com/crate-ci/typos/issues/779
           files: |

--- a/flake.nix
+++ b/flake.nix
@@ -32,6 +32,7 @@
               # Needed in darwin
               gnumake
               coreutils
+              pkg-config
 
               dprint
               tree

--- a/flake.nix
+++ b/flake.nix
@@ -26,6 +26,9 @@
               # https://github.com/kachick/irb-power_assert/issues/116
               # https://github.com/ruby/irb/pull/648
               libyaml
+              # Required to build ffi via steep dependency
+              # https://github.com/kachick/ruby-ulid/issues/423
+              libffi
 
               dprint
               tree

--- a/flake.nix
+++ b/flake.nix
@@ -29,6 +29,9 @@
               # Required to build ffi via steep dependency
               # https://github.com/kachick/ruby-ulid/issues/423
               libffi
+              # Needed in darwin
+              gnumake
+              coreutils
 
               dprint
               tree

--- a/flake.nix
+++ b/flake.nix
@@ -29,9 +29,7 @@
               # Required to build ffi via steep dependency
               # https://github.com/kachick/ruby-ulid/issues/423
               libffi
-              # Needed in darwin
-              gnumake
-              coreutils
+              # Needed for libffi in darwin
               pkg-config
 
               dprint

